test/integration/event_page_test.rb

```diff
@@ -4,59 +4,44 @@
   def setup
     @user = make_user
     @admin = make_admin
+    @event = make_event(organizer_id: @user.id, approved: true)
   end
 
-  test 'shows a logo image if the event has a logo' do
+  test 'does not show a Delete button if the Event is still open' do
     sign_in_as_user
-    make_event(name: 'The Event', approved: true, logo: 'logo-image')
-
-    visit events_path
-
-    assert page.text.include?('The Event')
-    assert_match /logo-image/, page.body
-    assert_no_match /event-default-.*\.png/, page.body
+    visit event_path(@event.id)
+    assert_not page.has_link?("Delete")
   end
 
-  test 'shows a default image if the event does not have a logo' do
+  test 'shows a Delete button if the Event has passed and the user is the organizer' do
     sign_in_as_user
-    make_event(name: 'The Event', approved: true)
-
-    visit events_path
-
-    assert page.text.include?('The Event')
-    assert_match /event-default-.*\.png/, page.body
+    @event.update_attributes(start_date: 1.week.ago, end_date: 1.day.ago, deadline: 1.week.ago)
+    visit event_path(@event.id)
+    assert page.has_link?("Delete")
   end
 
-  test 'shows Your events in the breadcrumb if the user is an organizer' do
-    sign_in_as_user
-    make_event(name: 'The Event', approved: true, organizer_id: @user.id)
-
-    visit events_path
-
-    click_link 'The Event'
-
-    assert page.text.include?('Your Events')
+  test 'does not show a Delete button if the user is not organizer' do
+    @event.update_attributes(start_date: 1.week.ago, end_date: 1.day.ago, deadline: 1.week.ago)
+    visit event_path(@event.id)
+    assert_not page.has_link?("Delete")
   end
 
-  test 'shows Events in the breadcrumb if the user is not an organizer' do
+  test 'updates events attributes after clicking "Delete" button' do
     sign_in_as_user
-    make_event(name: 'The Event', approved: true)
+    application = make_application(@event)
 
-    visit events_path
 
-    click_link 'The Event'
-
-    assert page.text.include?('Events')
-  end
+    @event.update_attributes(start_date: 1.week.ago, end_date: 1.day.ago, deadline: 1.week.ago)
 
-  test 'shows Admin in the breadcrumb if the user is an Admin' do
-    sign_in_as_admin
-    make_event(name: 'The Event', approved: true, organizer_id: @user.id)
+    visit event_path(@event.id)
 
-    visit events_path
+    assert page.has_link?("Delete")
+    click_link("Delete")
 
-    click_link 'The Event'
+    @event.reload
+    application.reload
 
-    assert page.text.include?('Admin')
+    assert_nil @event.name
```

Comment:
see above comment regarding controller test. why were the tests for breadcrumbs overwritten? 
We could instead test here for the redirect and flash if someone tries to view a deleted event?